python/vyos/ifconfig/ethernet.py

```diff
@@ -417,12 +418,38 @@ def set_ring_buffer(self, rx_tx, size):
             print(f'could not set "{rx_tx}" ring-buffer for {ifname}')
         return output
 
+    def set_switchdev(self, enable):
+        ifname = self.config['ifname']
+        if not bool(re.match(r"^eth[0-9]+$", ifname)):
+            print(f'switchdev mode is only possible on physical interfaces!')
+            return
+
+        addr, code = self._popen(f'ethtool -i {ifname} | grep bus-info | awk \'{{print $2}}\'')
```

Comment:
Will it be better to extend `Ethtool`  to get bus information?
https://github.com/vyos/vyos-1x/blob/current/python/vyos/ethtool.py
```
>>> from vyos.ethtool import Ethtool
>>> eth = Ethtool('eth0')
>>> eth.get_driver_name()
'virtio_net'
>>> 

Something like `get_bus_info` or `get_cpi_address`
```

Reply:
makes sense, will move it there